### PR TITLE
cmd: set double quotes as code delimiter

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/docs/pull/16093#issuecomment-1304582907

**- What I did**

Set `"` as code delimiter for flags usage output to fix issue with docs generation.

Also generates markdown in reference docs to be aligned as it seems we drifted a bit. Frontmatter section in md files doesn't seem to be used anymore so remove it as well.

~And I found out that https://github.com/docker/cli/pull/2936 and https://github.com/docker/cli/pull/2024 break markdown tables (see last 2 commits).~

In follow-up we should also validate docs like it's done in buildx: https://github.com/docker/buildx/blob/88852e2330a200d495342e01783a922a5328d55c/hack/dockerfiles/docs.Dockerfile#L30-L43

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by crazy-max** [comment]
@thaJeztah I wonder if https://github.com/docker/cli/blob/master/docs/reference/commandline/dockerd.md should not be in moby repo instead?

File: cli/flags/options.go
Comment (thaJeztah) [inline cli/flags/options.go:28]: Hm... I wonder if we can fix this in some other way. I know links in the output tend to be error-prone (users copy the link, and formatting code with it), but also to provide at least the "basic" information in the output.

Could we adjust the generator to replace newlines with `<br />` ?

**Comment by thaJeztah** [comment]
> I wonder if https://github.com/docker/cli/blob/master/docs/reference/commandline/dockerd.md should not be in moby repo instead?

Yeah, there was a bit of a debate at the time about that. I suggested it should be in the moby repo, but ISTR there were some reasons at the time to move it as well (but it may have been "it's easier to move all files").

We should look at moving it back, but ideally preserving history (might be fun as it will contain history from "before" it was moved and "after", and possibly because of the move; same history twice 😅 😓).

File: cli/flags/options.go
Comment (crazy-max) [reply cli/flags/options.go:28]: > Could we adjust the generator to replace newlines with `<br />` ?

Sounds good :+1:

Comment (crazy-max) [reply cli/flags/options.go:28]: Opened https://github.com/docker/cli-docs-tool/pull/33 to fix this.

File: docs/reference/commandline/config_inspect.md
Comment (thaJeztah) [inline docs/reference/commandline/config_inspect.md:11]: I opened https://github.com/docker/cli-docs-tool/pull/34 to make these tables more readable in source 😅

**Review by thaJeztah — COMMENTED** [review]
Thanks! Overall looking great! Left two comments (and opened a PR 😄 )

File: docs/reference/commandline/help.md
Comment (crazy-max) [inline docs/reference/commandline/help.md:10]: @thaJeztah Looks like we missed a case for subcommands in https://github.com/docker/cli-docs-tool/pull/34

Comment (thaJeztah) [reply docs/reference/commandline/help.md:10]: @crazy-max that's weird; all other commands do have the table properly aligned, except for `help`? 🤔

Comment (thaJeztah) [reply docs/reference/commandline/help.md:10]: Could it be using Cobra's built-in markdown generator? I don't see `help` in the list of things generated;

<details>

```
+ /tmp/docsgen --formats md --source /go/src/github.com/docker/cli/docs/reference/commandline --target /go/src/github.com/docker/cli/docs/reference/commandline
INFO: Generating Markdown for "docker attach"
INFO: Generating Markdown for "docker build"
INFO: Generating Markdown for "docker builder build"
INFO: Generating Markdown for "docker builder prune"
INFO: Generating Markdown for "docker builder"
INFO: Generating Markdown for "docker checkpoint create"
INFO: Generating Markdown for "docker checkpoint ls"
INFO: Generating Markdown for "docker checkpoint rm"
INFO: Generating Markdown for "docker checkpoint"
INFO: Generating Markdown for "docker commit"
INFO: Generating Markdown for "docker config create"
INFO: Generating Markdown for "docker config inspect"
INFO: Generating Markdown for "docker config ls"
INFO: Generating Markdown for "docker config rm"
INFO: Generating Markdown for "docker config"
INFO: Generating Markdown for "docker container attach"
INFO: Generating Markdown for "docker container commit"
INFO: Generating Markdown for "docker container cp"
INFO: Generating Markdown for "docker container create"
INFO: Generating Markdown for "docker container diff"
INFO: Generating Markdown for "docker container exec"
INFO: Generating Markdown for "docker container export"
INFO: Generating Markdown for "docker container inspect"
INFO: Generating Markdown for "docker container kill"
INFO: Generating Markdown for "docker container logs"
INFO: Generating Markdown for "docker container ls"
INFO: Generating Markdown for "docker container pause"
INFO: Generating Markdown for "docker container port"
INFO: Generating Markdown for "docker container prune"
INFO: Generating Markdown for "docker container rename"
INFO: Generating Markdown for "docker container restart"
INFO: Generating Markdown for "docker container rm"
INFO: Generating Markdown for "docker container run"
INFO: Generating Markdown for "docker container start"
INFO: Generating Markdown for "docker container stats"
INFO: Generating Markdown for "docker container stop"
INFO: Generating Markdown for "docker container top"
INFO: Generating Markdown for "docker container unpause"
INFO: Generating Markdown for "docker container update"
INFO: Generating Markdown for "docker container wait"
INFO: Generating Markdown for "docker container"
INFO: Generating Markdown for "docker context create"
INFO: Generating Markdown for "docker context export"
INFO: Generating Markdown for "docker context import"
INFO: Generating Markdown for "docker context inspect"
INFO: Generating Markdown for "docker context ls"
INFO: Generating Markdown for "docker context rm"
INFO: Generating Markdown for "docker context show"
INFO: Generating Markdown for "docker context update"
INFO: Generating Markdown for "docker context use"
INFO: Generating Markdown for "docker context"
INFO: Generating Markdown for "docker cp"
INFO: Generating Markdown for "docker create"
INFO: Generating Markdown for "docker diff"
INFO: Generating Markdown for "docker events"
INFO: Generating Markdown for "docker exec"
INFO: Generating Markdown for "docker export"
INFO: Generating Markdown for "docker history"
INFO: Generating Markdown for "docker image build"
INFO: Generating Markdown for "docker image history"
INFO: Generating Markdown for "docker image import"
INFO: Generating Markdown for "docker image inspect"
INFO: Generating Markdown for "docker image load"
INFO: Generating Markdown for "docker image ls"
INFO: Generating Markdown for "docker image prune"
INFO: Generating Markdown for "docker image pull"
INFO: Generating Markdown for "docker image push"
INFO: Generating Markdown for "docker image rm"
INFO: Generating Markdown for "docker image save"
INFO: Generating Markdown for "docker image tag"
INFO: Generating Markdown for "docker image"
INFO: Generating Markdown for "docker images"
INFO: Generating Markdown for "docker import"
INFO: Generating Markdown for "docker info"
INFO: Generating Markdown for "docker inspect"
INFO: Generating Markdown for "docker kill"
INFO: Generating Markdown for "docker load"
INFO: Generating Markdown for "docker login"
INFO: Generating Markdown for "docker logout"
INFO: Generating Markdown for "docker logs"
INFO: Generating Markdown for "docker manifest annotate"
INFO: Generating Markdown for "docker manifest create"
INFO: Generating Markdown for "docker manifest inspect"
INFO: Generating Markdown for "docker manifest push"
INFO: Generating Markdown for "docker manifest rm"
INFO: Generating Markdown for "docker manifest"
INFO: Generating Markdown for "docker network connect"
INFO: Generating Markdown for "docker network create"
INFO: Generating Markdown for "docker network disconnect"
INFO: Generating Markdown for "docker network inspect"
INFO: Generating Markdown for "docker network ls"
INFO: Generating Markdown for "docker network prune"
INFO: Generating Markdown for "docker network rm"
INFO: Generating Markdown for "docker network"
INFO: Generating Markdown for "docker node demote"
INFO: Generating Markdown for "docker node inspect"
INFO: Generating Markdown for "docker node ls"
INFO: Generating Markdown for "docker node promote"
INFO: Generating Markdown for "docker node ps"
INFO: Generating Markdown for "docker node rm"
INFO: Generating Markdown for "docker node update"
INFO: Generating Markdown for "docker node"
INFO: Generating Markdown for "docker pause"
INFO: Generating Markdown for "docker plugin create"
INFO: Generating Markdown for "docker plugin disable"
INFO: Generating Markdown for "docker plugin enable"
INFO: Generating Markdown for "docker plugin inspect"
INFO: Generating Markdown for "docker plugin install"
INFO: Generating Markdown for "docker plugin ls"
INFO: Generating Markdown for "docker plugin push"
INFO: Generating Markdown for "docker plugin rm"
INFO: Generating Markdown for "docker plugin set"
INFO: Generating Markdown for "docker plugin upgrade"
INFO: Generating Markdown for "docker plugin"
INFO: Generating Markdown for "docker port"
INFO: Generating Markdown for "docker ps"
INFO: Generating Markdown for "docker pull"
INFO: Generating Markdown for "docker push"
INFO: Generating Markdown for "docker rename"
INFO: Generating Markdown for "docker restart"
INFO: Generating Markdown for "docker rm"
INFO: Generating Markdown for "docker rmi"
INFO: Generating Markdown for "docker run"
INFO: Generating Markdown for "docker save"
INFO: Generating Markdown for "docker search"
INFO: Generating Markdown for "docker secret create"
INFO: Generating Markdown for "docker secret inspect"
INFO: Generating Markdown for "docker secret ls"
INFO: Generating Markdown for "docker secret rm"
INFO: Generating Markdown for "docker secret"
INFO: Generating Markdown for "docker service create"
INFO: Generating Markdown for "docker service inspect"
INFO: Generating Markdown for "docker service logs"
INFO: Generating Markdown for "docker service ls"
INFO: Generating Markdown for "docker service ps"
INFO: Generating Markdown for "docker service rm"
INFO: Generating Markdown for "docker service rollback"
INFO: Generating Markdown for "docker service scale"
INFO: Generating Markdown for "docker service update"
INFO: Generating Markdown for "docker service"
INFO: Generating Markdown for "docker stack config"
INFO: Generating Markdown for "docker stack deploy"
INFO: Generating Markdown for "docker stack ls"
INFO: Generating Markdown for "docker stack ps"
INFO: Generating Markdown for "docker stack rm"
INFO: Generating Markdown for "docker stack services"
INFO: Generating Markdown for "docker stack"
INFO: Generating Markdown for "docker start"
INFO: Generating Markdown for "docker stats"
INFO: Generating Markdown for "docker stop"
INFO: Generating Markdown for "docker swarm ca"
INFO: Generating Markdown for "docker swarm init"
INFO: Generating Markdown for "docker swarm join"
INFO: Generating Markdown for "docker swarm join-token"
INFO: Generating Markdown for "docker swarm leave"
INFO: Generating Markdown for "docker swarm unlock"
INFO: Generating Markdown for "docker swarm unlock-key"
INFO: Generating Markdown for "docker swarm update"
INFO: Generating Markdown for "docker swarm"
INFO: Generating Markdown for "docker system df"
INFO: Generating Markdown for "docker system dial-stdio"
INFO: Generating Markdown for "docker system events"
INFO: Generating Markdown for "docker system info"
INFO: Generating Markdown for "docker system prune"
INFO: Generating Markdown for "docker system"
INFO: Generating Markdown for "docker tag"
INFO: Generating Markdown for "docker top"
INFO: Generating Markdown for "docker trust inspect"
INFO: Generating Markdown for "docker trust key generate"
INFO: Generating Markdown for "docker trust key load"
INFO: Generating Markdown for "docker trust key"
INFO: Generating Markdown for "docker trust revoke"
INFO: Generating Markdown for "docker trust sign"
INFO: Generating Markdown for "docker trust signer add"
INFO: Generating Markdown for "docker trust signer remove"
INFO: Generating Markdown for "docker trust signer"
INFO: Generating Markdown for "docker trust"
INFO: Generating Markdown for "docker unpause"
INFO: Generating Markdown for "docker update"
INFO: Generating Markdown for "docker version"
INFO: Generating Markdown for "docker volume create"
INFO: Generating Markdown for "docker volume inspect"
INFO: Generating Markdown for "docker volume ls"
INFO: Generating Markdown for "docker volume prune"
INFO: Generating Markdown for "docker volume rm"
INFO: Generating Markdown for "docker volume update"
INFO: Generating Markdown for "docker volume"
INFO: Generating Markdown for "docker wait"
INFO: Generating Markdown for "docker"
+ clean
+ rm -rf /tmp/docker-cli-docsgen.ZpFcN9cDf8
```

</details>

Comment (crazy-max) [reply docs/reference/commandline/help.md:10]: Ah good point yeah, looks handled by Cobra. I guess it's fine then.

Comment (thaJeztah) [reply docs/reference/commandline/help.md:10]: I'm actually wondering if we should skip generating for the `help` command; we don't print it on the `--help` output (I think), and we never advertised it (I think it's one of those automatic features from Cobra). 🤔

File: cli/command/container/update.go
Comment (thaJeztah) [inline cli/command/container/update.go:60]: Not something new, but my eye dropped on these in the new tables; we have various options where `0` (or `0 sec`) is the default.

In the case above, "0" is explained (so "OK(ish)") but in various other cases, `0` very likely means "use default" (e.g. `cpu-period`). We should review which ones need "0" to be shown as default value, and for which ones we should (ideally) _not_ print the default (as the default basically means "not set").

File: cli/command/node/update.go
Comment (thaJeztah) [inline cli/command/node/update.go:32]: I'm slightly on the fence on these (but admitted: we weren't very consistent);  slightly wondering what's clearer on intent;

- `"foo"|"bar"` either `"foo"` (string) or `"bar"` (string)
- `"foo|bar"` a string of either foo or bar
- some other notation? (`"foo", "bar", or "baz"`) ?
- extended notation for all of these (multiple lines; short explanation for each option)


/cc @dvdksn @vvoland @usha-mandya

Comment (dvdksn) [reply cli/command/node/update.go:32]: in the [build docs](https://docs.docker.com/build/cache/backends/local/) we've used comma-separated notation, and the options wrapped in code spans. My subjective opinion is that comma is more readable than pipes, but this is probably debatable.

Comment (vvoland) [reply cli/command/node/update.go:32]: IMO single quotes containing all options are confusing (`"foo|bar"`). 
Personally I find commas clearer than pipes when only one value can be given at once, so 3rd option looks good for me:

```go
flags.StringVar(&options.role, flagRole, "", `Role of the node ("worker" or "manager")`)
flags.StringVar(&options.availability, flagAvailability, "", `Availability of the node ("active", "pause" or "drain")`)
```

**Comment by thaJeztah** [comment]
Looks like an e2e test needs updating;

```
=== FAIL: e2e/stack TestStackDeployHelp (0.05s)
    help_test.go:13: assertion failed: 
        --- expected
        +++ actual
        @@ -12,6 +12,6 @@
         ······--prune··················Prune·services·that·are·no·longer·referenced
         ······--resolve-image·string···Query·the·registry·to·resolve·image·digest
        -·······························and·supported·platforms
        -·······························("always",·"changed",·"never")·(default·"always")
        +·······························and·supported·platforms·("always",
        +·······························"changed",·"never")·(default·"always")
         ······--with-registry-auth·····Send·registry·authentication·details·to
         ·······························Swarm·agents
        
        
        You can run 'go test . -update' to automatically update testdata/stack-deploy-help.golden to the new expected value.'
```

File: docs/reference/commandline/docker.md
Comment (thaJeztah) [inline docs/reference/commandline/docker.md:7]: Looks like renaming `cli.md` to `docker.md` doesn't work currently; `docker.yaml` does not include the content; I opened a draft PR to test this PR; https://github.com/docker/docs/pull/16416


Maybe we should have a look at this separately; also to see what we want to split to what page (some of this could go into `index.md`, other bits could go into `../run.md`, and other parts into `docker.md`.

**Review by albers — COMMENTED** [review]
I was invited for review as a code owner of the bash completion, so I'm not an expert in most areas touched by this PR.
As far as I can tell, the changes look sound, thanks.
I found an incorrect docs reference, please update it.

File: docs/reference/commandline/dockerd.md
Comment (thaJeztah) [inline docs/reference/commandline/dockerd.md:16]: Looks like some accidental whitespace changes

File: docs/reference/commandline/cli.md
Comment (thaJeztah) [inline docs/reference/commandline/cli.md:20]: Here as well (comment and redirects_from) 🙈

File: docs/reference/commandline/docker.md
Comment (thaJeztah) [inline docs/reference/commandline/docker.md:20]: Now that we (for now) are keeping the `cli.md`, this file should not have these redirects, and not have the content we copied from `cli.md`

**Review by thaJeztah — APPROVED** [review]
LGTM!

I swapped the last two commits, so that we have the option of cherry-picking the anchor quote fixes

**Comment by thaJeztah** [comment]
Arf, and of course then it wants the DCO to be fixed; let me do so

**Pull request merged**
